certs/certs.go

```diff
@@ -100,9 +112,86 @@ func ValidateRoot(certStore trustmanager.X509Store, root *data.Signed, gun strin
 		}
 	} else {
 		logrus.Debugf("found no currently valid root certificates for %s", gun)
+		logrus.Debugf("using trust_pinning config to bootstrap trust: %v", trustPinning)
```

Comment:
Agreed, `ValidateRoot` is giant and this control flow can be factored out.  I'll work on refactoring this.